src/Proxy/LazyServiceFactory.php

```diff
@@ -42,8 +43,14 @@ public function __invoke(
     ): VirtualProxyInterface {
         if (isset($this->servicesMap[$name])) {
             $initializer = static function (&$wrappedInstance, LazyLoadingInterface $proxy) use ($callback): bool {
+                $initializer = $proxy->getProxyInitializer();
                 $proxy->setProxyInitializer(null);
-                $wrappedInstance = $callback();
+                try {
+                    $wrappedInstance = $callback();
+                } catch (Throwable $e) {
+                    $proxy->setProxyInitializer($initializer);
+                    throw $e;
```

Comment:
What if we just swap these two rows: `$proxy->setProxyInitializer(null);` and `$wrappedInstance = $callback();`?

It'll look like this:
```php
                $wrappedInstance = $callback();
                $proxy->setProxyInitializer(null);
```